lib/wings/transformer_value_mapper.rb

```diff
@@ -7,7 +7,28 @@ module Wings
   #
   # This top level matcher has registered several internal mappers which handle
   # indivdual value types from the source data.
-  class ValueMapper < ::Valkyrie::ValueMapper; end
+  class TransformerValueMapper < ::Valkyrie::ValueMapper; end
+
+  class NestedResourceMapper < ::Valkyrie::ValueMapper
+    # needs to register before the ResourceMapper or it will use that one
+    # instead
+    TransformerValueMapper.register(self)
+
+    def self.handles?(value)
+      value.is_a? Wings::ActiveFedoraConverter::NestedResource
+    end
+
+    def result
+      id = ActiveFedora::Base.uri_to_id(value.id)
+      default_obj = ActiveFedora::Base.find id
```

Comment:
This I am not sure about. It comes back as a `DefaultWork`. It's submitted as a hash, with the property defined as a `NestedResource`. Why doesn't it come back with that same model?

I think if it would come back right we wouldn't need to transform the attributes and create the nested_object.

Reply:
thanks @revgum for helping me sort this out and fix it up!